compiler/incremental-compilation-impl/test/org/jetbrains/kotlin/incremental/classpathDiff/ClasspathSnapshotterTest.kt

```diff
@@ -5,141 +5,93 @@
 
 package org.jetbrains.kotlin.incremental.classpathDiff
 
-import org.jetbrains.kotlin.incremental.classpathDiff.ClasspathSnapshotTestCommon.Util.snapshot
+import org.jetbrains.kotlin.incremental.classpathDiff.ClasspathSnapshotTestCommon.ClassFileUtil.readBytes
+import org.jetbrains.kotlin.incremental.classpathDiff.ClasspathSnapshotTestCommon.SourceFile.JavaSourceFile
+import org.jetbrains.kotlin.incremental.classpathDiff.ClasspathSnapshotTestCommon.SourceFile.KotlinSourceFile
 import org.junit.Assert.assertEquals
 import org.junit.Assert.assertNotEquals
-import org.junit.Before
 import org.junit.Test
-import org.junit.runner.RunWith
-import org.junit.runners.Parameterized
 import java.io.File
 
 abstract class ClasspathSnapshotterTest : ClasspathSnapshotTestCommon() {
 
-    protected abstract val testSourceFile: ChangeableTestSourceFile
-
-    private lateinit var testClassSnapshot: ClassSnapshot
-
-    @Before
-    fun setUp() {
-        testClassSnapshot = testSourceFile.compileAndSnapshot()
-    }
-
-    @Test
-    fun `test ClassSnapshotter's result against expected snapshot`() {
-        assertEquals(getExpectedSnapshotFile().readText(), testClassSnapshot.toGson())
-    }
-
-    private fun getExpectedSnapshotFile() = testSourceFile.sourceFile.asFile().path.let {
-        File(it.substringBeforeLast("src") + "expected-snapshot" + it.substringAfterLast("src").substringBeforeLast('.') + ".json")
-    }
-
-    @Test
-    fun `test ClassSnapshotter extracts ABI info from a class`() {
-        // Change public method signature
-        val updatedSnapshot = testSourceFile.changePublicMethodSignature().compileAndSnapshot()
-
-        // The snapshot must change
-        assertNotEquals(testClassSnapshot.toGson(), updatedSnapshot.toGson())
-    }
-
-    @Test
-    fun `test ClassSnapshotter does not extract non-ABI info from a class`() {
-        // Change method implementation
-        val updatedSnapshot = testSourceFile.changeMethodImplementation().compileAndSnapshot()
-
-        // The snapshot must not change
-        assertEquals(testClassSnapshot.toGson(), updatedSnapshot.toGson())
-    }
-}
-
-class KotlinClassesClasspathSnapshotterTest : ClasspathSnapshotterTest() {
-    override val testSourceFile = SimpleKotlinClass(tmpDir)
-}
-
-@RunWith(Parameterized::class)
-class JavaClassesClasspathSnapshotterTest(private val protoBased: Boolean) : ClasspathSnapshotTestCommon() {
-
     companion object {
-        @Parameterized.Parameters(name = "protoBased={0}")
-        @JvmStatic
-        fun parameters() = listOf(true, false)
+        val testDataDir =
```

Comment:
It's used by `Kotlin/JavaOnlyClasspathSnapshotterTest`, which is in the same file but outside this class. I can make it a private top-level constant, but it will be outside this class. I slightly prefer the current position, but let me know if you'd prefer the other way.